plugins/modules/stop_region.py

```diff
@@ -267,32 +282,42 @@ def __init__(self):
         self.msg = ""
 
     def main(self):
-        if self._module.params.get(SDTRAN):
-            self._validate_sdtran(self._module.params[SDTRAN])
-        if not self._module.params.get(JOB_ID) and not self._module.params.get(JOB_NAME):
-            self._fail("At least one of {0} or {1} must be specified".format(
-                JOB_ID, JOB_NAME))
-        self.result = self.get_result()
-        self._module.exit_json(**self.result)
-
-    def _validate_sdtran(self, transaction):  # type: (str) -> None
-        if len(transaction) > 4:
-            self._fail(
-                "Value: {0}, is invalid. SDTRAN value must be  1-4 characters.".format(transaction))
-
-    def _fail(self, msg):  # type: (str) -> None
-        self.failed = True
-        self.msg = msg
-        self.result = self.get_result()
-        self._module.fail_json(**self.result)
-
-    def get_result(self):  # type: () -> dict
-        return {
-            "changed": False,
-            "failed": self.failed,
-            "executions": [],
-            "msg": self.msg
-        }
+        # At this point, this module only gets executed with JOB_ID
+        # This is as a wrapper to jls via ibm_zos_core job to clean-up the output
+        # if there is no job found with that ID (ZOAU throws an exception)
+        job_id = self._module.params.get(JOB_ID)
+
+        jobs_raw: list[dict] = get_jobs_wrapper(job_id)
+
+        if not jobs_raw:
+            self._module.fail_json("No jobs found with id {0}".format(job_id))
+
+        if len(jobs_raw) > 1:
+            self._module.fail_json("Multiple jobs found with ID {0}".format(job_id))
+
+        job = jobs_raw[0]
+
+        if not job.get("job_id") == job_id:
+            self._module.fail_json("Couldn't determine job name for job ID {0}".format(job_id))
+
+        job_name = job.get("job_name")
+        if not job_name:
+            self._module.fail_json("Couldn't determine job name for job ID {0}".format(job_id))
+
+        ret_code = job.get("ret_code")
+        if not ret_code:
+            self._module.fail_json("Couldn't determine status for job ID {0} with name {1}".format(job_id, job_name))
+
+        status = ret_code.get("msg")
+        if not status:
+            self._module.fail_json("Couldn't determine status for job ID {0} with name {1}".format(job_id, job_name))
```

Comment:
As your using the exact same error message for these 2 cases, wondering if you could condense it to something like:

```python
if not (job.get("ret_code") or job["ret_code"].get("msg")):
  self._module.fail_json("Couldn't determine status for job ID {0} with name {1}".format(job_id, job_name))
```